test/tests/functional/pbs_mom_dynamic_resource_ext.py

```diff
@@ -0,0 +1,212 @@
+
+        (fd, fp) = self.du.mkstemp(prefix="mom_dyn_resc", suffix=".scr",
+                                   body=script_body)
+        self.du.chmod(path=fp, mode=0755)
+        os.close(fd)
+
+        mom_config_str = "!" + fp
+        self.mom.add_config({resc_name: mom_config_str})
+        self.scheduler.set_sched_config({'mom_resources': '"foo"'},
+                                        validate=True)
+        self.scheduler.add_resource(resc_name)
+
+    def test_t1(self):
+        """
+        Test for host level consumable size resource
+        with string mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "size", "nh", "/bin/echo abc")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 1}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (1kb != (null))"
+        self.scheduler.log_match(c)
+
+    def test_t2(self):
+        """
+        Test for host level consumable float resource
+        with string mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "float", "nh", "/bin/echo abc")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 2}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (2 != (null))"
+        self.scheduler.log_match(c)
+
+    def test_t3(self):
+        """
+        Test for host level string resource
+        with integer mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "string", "h", "/bin/echo 3")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 'abc'}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (abc != 3)"
+        self.scheduler.log_match(c)
+
+    def test_t4(self):
+        """
+        Test for host level string_array resource
+        with float mom dynamic value
```

Comment:
I don't think you are not really testing what you think you are testing.  What you are testing is that the scheduler is correctly not matching the string 'red' and the string '3.14'.